src/zephyr_coap_req.c

```diff
@@ -544,7 +544,7 @@ int golioth_coap_req_cb(struct golioth_client *client,
     if (err)
     {
         LOG_ERR("Failed to create new CoAP GET request: %d", err);
-        goto free_req;
```

Comment:
We can do that as a "bug prevention" I think. Normally the most common practice and convention is to not rely on any "out" parameters when the function returns error code.

I have 50/50 thoughts on that. So whatever you lean to / prefer. I can create a patch if you think it will be valueable.